pkg/addon/utils.go

```diff
@@ -504,7 +504,9 @@ func checkBondComponentExist(u unstructured.Unstructured, app v1beta1.Applicatio
 }
 
 // FilterDependencyRegistries will return all registries besides the target registry itself
-func FilterDependencyRegistries(i int, registries []Registry) []Registry {
+func FilterDependencyRegistries(i int, rs []Registry) []Registry {
+	registries := make([]Registry, len(rs))
+	copy(registries, rs)
 	if i < len(registries) {
```

Comment:
What if `i<0`?

Something like this will do (possibly copies fewer elements)

```go
func FilterDependencyRegistries(i int, rs []Registry) []Registry {
	if i >= len(rs) {
		return rs
	}
	if i < 0 {
		return rs
	}
	ret := make([]Registry, len(rs)-1)
	copy(ret, rs[:i])
	copy(ret[i:], rs[i+1:])
	return ret
}
```

Reply:
Good suggestion.